EIPS/eip-2678.md

```diff
@@ -0,0 +1,2132 @@
+<colgroup>
+<col style="width: 50%" />
+<col style="width: 50%" />
+</colgroup>
+<tbody>
+<tr class="odd">
+<td><p>Required</p></td>
+<td><p>No</p></td>
+</tr>
+<tr class="even">
+<td><p>Type</p></td>
+<td><p>Object</p></td>
+</tr>
+<tr class="odd">
+<td><p>Format</p></td>
+<td><p><strong>Must</strong> conform to <a href="#bytecode-object">the Bytecode Object</a> format.</p></td>
+</tr>
+</tbody>
+</table>
+
+
+Every entry in the `linkReferences` for this bytecode **must** have a
+corresponding entry in the `linkDependencies` section.
+
+
+<div id="compiler-information-object"></div>
+
+The *Compiler Information* Object
+=================================
+
+The `compilers` field defines the various compilers and settings used
+during compilation of any [Contract Types](#term-contract-type) or
+[Contract Instance](#term-contract-instance) included in this package.
+
+A *Compiler Information* object is defined to have the following
+key/value pairs.
+
+#### Name: `name`
+
+The `name` field defines which compiler was used in compilation.
+
+<table>
+<colgroup>
+<col style="width: 50%" />
+<col style="width: 50%" />
+</colgroup>
+<tbody>
+<tr class="odd">
+<td><p>Required</p></td>
+<td><p>Yes</p></td>
+</tr>
+<tr class="even">
+<td><p>Key</p></td>
+<td><p><code>name</code></p></td>
+</tr>
+<tr class="odd">
+<td><p>Type</p></td>
+<td><p>String</p></td>
+</tr>
+</tbody>
+</table>
+
+
+#### Version: `version`
+
+The `version` field defines the version of the compiler. The field
+**should** be OS agnostic (OS not included in the string) and take the
+form of either the stable version in
+[semver](http://semver.org/) format or if built on a
+nightly should be denoted in the form of `<semver>-<commit-hash>` ex:
+`0.4.8-commit.60cc1668`.
+
+<table>
+<colgroup>
+<col style="width: 50%" />
+<col style="width: 50%" />
+</colgroup>
+<tbody>
+<tr class="odd">
+<td><p>Required</p></td>
+<td><p>Yes</p></td>
+</tr>
+<tr class="even">
+<td><p>Key</p></td>
+<td><p><code>version</code></p></td>
+</tr>
+<tr class="odd">
+<td><p>Type</p></td>
+<td><p>String</p></td>
+</tr>
+</tbody>
+</table>
+
+
+#### Settings: `settings`
+
+The `settings` field defines any settings or configuration that was used
+in compilation. For the `"solc"` compiler, this **should** conform to
+the [Compiler Input and Output
+Description](http://solidity.readthedocs.io/en/latest/using-the-compiler.html#compiler-input-and-output-json-description).
+
+<table>
+<colgroup>
+<col style="width: 50%" />
+<col style="width: 50%" />
+</colgroup>
+<tbody>
+<tr class="odd">
+<td><p>Required</p></td>
+<td><p>No</p></td>
+</tr>
+<tr class="even">
+<td><p>Key</p></td>
+<td><p><code>settings</code></p></td>
+</tr>
+<tr class="odd">
+<td><p>Type</p></td>
+<td><p>Object</p></td>
+</tr>
+</tbody>
+</table>
+
+
+#### Contract Types: `contractTypes`
+
+A list of the [Contract Alias](#term-contract-alias) or [Contract Types](#term-contract-type) in this package
+that used this compiler to generate its outputs.
```

Comment:
Why is this done via `CompilerInformation` -> `ContractType`, instead of the other way around? Would make more sense if `ContractType` linked to `CompilerInformation` object instead, ensuring no duplication

Reply:
Probably, but given that `Compilers` is an array of `CompilerInformationObject`s, I'm not sure of any reliable method to link `ContractType` -> `CompilerInformation`.

Reply:
By index? Both should be handled by a framework, so I see no issue in them being able to rectify that

Reply:
I feel somewhat uneasy about relying on the index. It doesn't seem like a reliable strategy to maintain the links, and it's not as explicit as the current strategy, which while clunky in terms of enforcing no duplication, it's more obvious how the links work without a thorough knowledge of the spec.